drivers/hmis_external_apis/public/schemas/form_definition.json

```diff
@@ -203,84 +829,269 @@
             "HOH_AND_ADULTS",
             "VETERAN_HOH"
           ]
+        }
+      }
+    },
+    "groupFormNode": {
+      "$comment": "Group form nodes",
+      "allOf": [
+        {
+          "$ref": "#/$defs/baseFormNode"
         },
-        "type": {
+        {
+          "$ref": "#/$defs/dataCollectedAbout"
+        },
+        {
+          "type": "object",
+          "$comment": "child item array should be required, but cant because when you add a new group to the form builder its empty and we validate on save. maybe add an exception?",
+          "required": [
+          ],
```

Comment:
Note to self to update/remove this comment. @ttoomey it's correct that groups _should_ have children, but the Form Builder needs to maintain the ability to add a group, save, and then add children to it.

The form renderer is capable of rendering childless groups, so I think it's OK to keep them un-required.

Reply:
The point of this is to ensure that our frontend form builder is only producing valid shapes. So I definitely think we should validate the schema when you save changes to an item (on a draft). If we don't, it's possible our form builder is allowing you to create invalid forms, and then when you go to publish you're faced with a bunch of (possibly un-fixable) errors and need to track them down. It will also make QA a lot harder if we can't surface issues until publish. So I would like to keep validation on save.

Requiring children in groups is the only thing I can think of to omit from the draft stage. As mentioned I don't feel like it's important enough to warrant building a specific workaround check for it. (Basically, nothing 'bad' happens from having an empty group)